api/src/main/java/info/rmapproject/api/auth/ApiUserServiceImpl.java

```diff
@@ -64,6 +69,7 @@ public AuthorizationPolicy getCurrentAuthPolicy() throws RMapApiException {
 	    if (authorizationPolicy == null) {
 	        throw new RMapApiException(ErrorCode.ER_COULD_NOT_RETRIEVE_AUTHPOLICY);
 	        }
+	    log.debug("Authorization policy retrieved with username " + ((authorizationPolicy.getUserName()==null) ? "" : authorizationPolicy.getUserName()));
```

Comment:
No need to change this, but just want to point out that SLF4J can parameterize logging statements:
`log.debug("User {} performed action {}", user, action)`

The advantage is that the parameters are only evaluated if the logging threshold is reached.  The alternate version:
`log.debug("User " + user + " performed action " + action)`
performs string concatenation regardless of whether the debug threshold is reached or not.

SLF4J is pretty smart; if a `Throwable` is the last parameter, it automatically prints a stacktrace:
`log.error("Encountered an error: {}", ex.getMessage(), ex)`

In this example there's only one parameter placeholder: for the exception message.  The `ex` parameter (the `Exception`) is automatically used to generate a stacktrace with the logging message.

Reply:
thanks, this is a good tip, I will change these for next update